src/group_theory/quotient_group.lean

```diff
@@ -412,4 +412,69 @@ monoid_hom.to_mul_equiv
 
 end third_iso_thm
 
+
+section trivial
+
+/-These first few lemmas should maybe go somewhere else, where?-/
+
+lemma subsingleton_quot_iff_forall {α : Type u} (r : α → α → Prop ) :
+  subsingleton (quot r) ↔ ∀ a b, eqv_gen r a b :=
+begin
+  rw subsingleton_iff,
+  refine (surjective_quot_mk _).forall.trans (forall_congr $ λ a, _),
+  refine (surjective_quot_mk _).forall.trans (forall_congr $ λ b, _),
+  exact quot.eq,
+end
```

Comment:
I think you should have a version of this lemma for `setoid` too, in `data/setoid/basic.lean` and something like
```lean
@[simp]
lemma setoid.rel_top {α : Type u} : (⊤ : setoid α).rel = ⊤ := rfl

lemma subsingleton_quotient_iff_eq_top {α : Type u} {s : setoid α} :
  subsingleton (_root_.quotient s) ↔ s = ⊤ :=
begin
  simp only [subsingleton_iff, eq_top_iff, setoid.le_def, setoid.rel_top, top_apply, forall_const],
  refine (surjective_quotient_mk _).forall.trans (forall_congr $ λ a, _),
  refine (surjective_quotient_mk _).forall.trans (forall_congr $ λ b, _),
  exact quotient.eq',
end
```

Reply:
Sure, I put this in that file. Is there a convention as to where in the file it should go? I just stuck it in at the end,